gpt4all-bindings/typescript/scripts/assert-backend-sources.js

```diff
@@ -0,0 +1,46 @@
+const fs = require("fs");
+const path = require("path");
+
+// Copies the shared llmodel sources from gpt4all-backend into the backend folder.
+// These are dependencies of the bindings and will be required in case node-gyp-build
+// cannot find a prebuild. This script is used in the package install hook and will
+// be executed BOTH when `yarn install` is run in the root folder AND when the package
+// is installed as a dependency in another project.
+
+const backendDeps = [
+    "llmodel.h",
+    "llmodel.cpp",
+    "llmodel_c.cpp",
+    "llmodel_c.h",
+    "sysinfo.h",
+    "dlhandle.h",
+];
+
+const sourcePath = path.resolve(__dirname, "../../../gpt4all-backend");
+const destPath = path.resolve(__dirname, "../backend");
+
+// Silently ignore if the backend sources are not available.
+// When the package is installed as a dependency, gpt4all-backend will not be present.
+if (fs.existsSync(sourcePath)) {
```

Comment:
Not what we want here I think, because when the script gets executed by the user during the install hook (ie user installs the package as a dependency) this would create a `gpt4all-backend` path under node_modules. And then error because there are no gpt4all-backend sources under that newly created dir. We really just want it to do nothing in that case.